### PR TITLE
2380 document featsecret introduce secret management

### LINKED ISSUE #2380: Document: feat(secret): introduce secret management
This issue tracks the documentation update needed for the merged PR #17456.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/17456
Source PR Merged At: 2024-07-15T16:48:41Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - Secret management feature
- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/17456

- **Related doc issue**
  
  Resolves  https://github.com/risingwavelabs/risingwave-docs/issues/2380

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

File: docs/deploy/manage-secrets.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+```
+
+## Use secrets as a file
+
+Some connectors need credentials stored as file paths (e.g., `ssl.ca.location`), where the file contains the secret. RisingWave allows you to reference a secret as a file path.
+
+```sql title="Reference a secret as a file path"
+CREATE TABLE district (
+    d_id INTEGER,
+    PRIMARY KEY (d_id, d_w_id)
```
Comment (yuhao-su) [inline docs/deploy/manage-secrets.md]:
```suggestion
    PRIMARY KEY (d_id)
```

**Review by yuhao-su — APPROVED** [review]
Rest LGTM

File: docs/deploy/manage-secrets.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+RisingWave provides four key secret management operations:
+
+- Creating secrets.
+
+- Using secrets.
+
+- Using secrets as a file.
+
+- Dropping secrets.
```
Comment (WanYixian) [inline docs/deploy/manage-secrets.md:28]:
```suggestion
- Create secrets.

- Use secrets.

- Use secrets as a file.

- Drop secrets.
```

File: docs/deploy/manage-secrets.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+CREATE SECRET mysql_pwd WITH (
+  backend = 'meta'
+) AS '123';
+```
+
+:::note
+Currently only the meta backend is supported.
+:::
+
+## Use Secrets
```
Comment (WanYixian) [inline docs/deploy/manage-secrets.md]:
```suggestion
## Use secrets
```

**Review by WanYixian — APPROVED** [review]
LGTM

File: docs/deploy/manage-secrets.md (hunk 2 above)
Comment (ShanlanLi) [reply docs/deploy/manage-secrets.md:28]: thanks, this is not a heading. So I use ing.

**Pull request merged**